# PRJNA622558 - Yeast pulldown compendium

This dataset contains paired-end Illumina data for a large number of ChIP-seq pulldowns in yeast.
It will be used to explore how to find "peaks" in sequencing data.snap

## Paper
Nature. Author manuscript; available in PMC 2021 Sep 10.
Nature. 2021 Apr; 592(7853): 309â€“314.
Published online 2021 Mar 10. doi: 10.1038/s41586-021-03314-8

A high-resolution protein architecture of the budding yeast genome
Matthew J. Rossi, Prashant K. Kuntala, William K.M. Lai, Naomi Yamada, Nitika Badjatia, Chitvan Mittal, Guray Kuzu, Kylie Bocklund, Nina P. Farrell, Thomas R. Blanda, Joshua D. Mairose, Ann V. Basting, Katelyn S. Mistretta, David J. Rocco, Emily S. Perkinson, Gretta D. Kellogg, Shaun Mahony, and B. Franklin Pugh

## Abstract

The genome-wide protein architecture of chromatin that maintains chromosome integrity and gene regulation is ill-defined.
Here we use ChIP-exo/seq to define this structure in Saccharomyces.
We identified 21 ensembles consisting of ~400 different proteins related to DNA replication, centromeres, subtelomeres, transposons, and RNA polymerase (Pol) I, II, and III transcription.
Replication proteins engulfed a nucleosome, centromeres lacked a nucleosome, and repressive proteins encompassed three nucleosomes at subtelomeric X-elements.
We find that most Pol II promoters evolved to lack a regulatory region, having only a core promoter.
These constitutive promoters comprised a short nucleosome-free region (NFR) adjacent to a +1 nucleosome, which together bound TFIID to form a preinitiation complex (PIC).
Positioned insulators protected core promoters from upstream events.
A small fraction of promoters were architected for inducibility, wherein sequence-specific transcription factors (TFs) create a nucleosome-depleted region (NDR) that is distinct from NFRs.
We describe TF structural interactions with the genome and cognate cofactors, including nucleosomal and transcriptional regulators RPD3-L, SAGA, NuA4, Tup1, Mediator, and SWI-SNF.
Surprisingly, we do not detect TF-TFIID interactions, suggesting that they do not stably occur.
Our model for gene induction involves TFs, cofactors, and general factors like TBP and TFIIB, but not TFIID.
However, constitutive transcription involves TFIID but not TFs and cofactors.
From this we define a highly integrated network of TF-regulated transcription.

## Dataset

https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE147927

This dataset contains paired-end Illumina data for a large number of pulldowns.
Raw data will be downloaded as needed.

```bash
/data/share/courses/OMICS/PRJNA622558/
```

There are preprocessed versions provided as well.
